qiskit_nature/circuit/library/ansatzes/succ_full.py

```diff
@@ -0,0 +1,208 @@
+        return self.generate_excitations(self.num_spin_orbitals, self.num_particles)
+
+    def generate_excitations(
+        self, num_spin_orbitals: int, num_particles: Tuple[int, int]
+    ) -> List[Tuple[Tuple[int, ...], Tuple[int, ...]]]:
+        """Generates the excitations for the SUCC_full Ansatz.
+        Args:
+            num_spin_orbitals: the number of spin orbitals.
+            num_particles: the number of alpha and beta electrons. Note, these must be identical for
+            this class.
+        Raises:
+            QiskitNatureError: if the number of alpha and beta electrons is not equal.
+        Returns:
+            The list of excitations encoded as tuples of tuples. Each tuple in the list is a pair of
+            tuples. The first tuple contains the occupied spin orbital indices whereas the second
+            one contains the indices of the unoccupied spin orbitals.
+        """
+        self._validate_num_particles(num_particles)
+
+        excitations: List[Tuple[Tuple[int, ...], Tuple[int, ...]]] = []
+        excitations.extend(
+            generate_fermionic_excitations(
+                1,
+                num_spin_orbitals,
+                num_particles,
+                alpha_spin=self.include_singles[0],
+                beta_spin=self.include_singles[1],
+            )
+        )
+
+        num_electrons = num_particles[0]
+
+        # generate alpha-spin and beta-spin orbital indices for occupied and unoccupied ones
+        alpha_excitations = get_alpha_excitations(
+            num_electrons, num_spin_orbitals, self._generalized
+        )
+        beta_excitations = get_beta_excitations(num_electrons, num_spin_orbitals, self._generalized)
+        logger.debug("Generated list of single alpha excitations: %s", alpha_excitations)
+        logger.debug("Generated list of single beta excitations: %s", beta_excitations)
+
+        # Find all possible double excitations constructed from the list of single excitations.
+        # Note, that we use `product` here, in order to also get those double
+        # excitations form an alpha excitation and a beta excitation. We will need those in the
+        # following post-processing step.
+        pool = itertools.product(alpha_excitations, beta_excitations)
+
+        for exc in pool:
+            # find the two excitations (Note: SUCCD only works for double excitations!)
+            alpha_exc, second_exc = exc[0], exc[1]
+            # shift the second excitation into the beta-spin orbital index range
+            beta_exc = (
+                second_exc[0],
+                second_exc[1],
+            )
+            # add the excitation tuple
+            occ: Tuple[int, ...]
+            unocc: Tuple[int, ...]
+            occ, unocc = zip(alpha_exc, beta_exc)
+            exc_tuple = (occ, unocc)
+            excitations.append(exc_tuple)
+            logger.debug("Added the excitation: %s", exc_tuple)
```

Comment:
This entire part is actually equivalent to the following:
```suggestion
        excitations.extend(
            generate_fermionic_excitations(
                2,
                num_spin_orbitals,
                num_particles,
            )
        )
```

Reply:
Hi morssinek, for this part, I would like to keep the code I used. Because I tried "**generate_fermionic_excitations( 2, ...)** "but it returns wrong excitation when there are two or more alpha/beta particles. For example, **num_particles,num_spin_orbitals=(2,2), 8**, the excitation_list will include **((0,1),(2,3))** the two excited particles are both alpha particles, which I think is wrong.

Reply:
Hm you're right that these excitations should be excluded from SUCCD. However, you can already achieve that with the following:
```python
generate_fermionic_excitations(
    2,
    num_spin_orbitals,
    num_particles,
    max_spin_excitation=1,
)
```